modules/calib3d/src/ap3p.cpp

```diff
@@ -45,7 +45,12 @@ void solveQuartic(const double *factors, double *realRoots) {
     complex<double> sqrt_2m = sqrt(static_cast<complex<double> >(-2 * p4 / 3 + t));
     double B_4A = -a3 / (4 * a4);
     double complex1 = 4 * p4 / 3 + t;
+#if defined(__clang__)
+    complex<double> complex2 = 2 * q4;
+    complex2 = complex<double>(complex2.real() / sqrt_2m.real(), 0);
```

Comment:
Can sqrt_2m have non-zero imagination? (from sqrt call of negative number)
Perhaps we need full formula for complex numbers division (and optimize it for zero `complex2.imag()`).

Reply:
Thank you for nice comment.
All the test case from gtest doesn't fail to 0, but AFAIK, there's no guarantee on this point.
Let me think of that point.

Reply:
As far as I understood, this function ```solveQuartic``` is calculating real roots from quartic equation.
Mathematically, there is a possibility that imaginary part becomes non-zero, (probably caused by wrong correspondence)
Should I add assertion before the ```sqrt``` or shall I keep as it is ?